### DISCUSSION TITLE
Pass jq's input on the command line?

### DISCUSSION THREAD

**Discussion by StephenWall** [stedolan/jq#3107] [Q&A] [answered]
Is there a way to pass `jq` it's input data on the command line rather than in a file or stdin?
For example, if I have a shell variable `data` that contains `{"name":"program","version":"1.2.3"}`, I could use `result=$(echo $data | jq -j '.name,"-",.version'` to get my desired result of `program-1.2.3`.  Is there a way to do this without the `echo` and the pipe?

**Comment by wader** [comment, marked as answer]
Yes, there is `--argjson name value`
```sh
$ jq -n --argjson data '{"name":"program","version":"1.2.3"}' '$data'
{
  "name": "program",
  "version": "1.2.3"
}
$ jq -n --argjson data '{"name":"program","version":"1.2.3"}' '$data.version'
"1.2.3"
```

**Comment by StephenWall** [comment]
Ah, I saw that [`--argjson`] in the man page, but couldn't get it to work. I needed the `-n`.  `jq -n -j --argjson data "$data" '$data.name,"-",$data.version'` works nicely.  Thank you.
